cairo/scripts/prove_block.py

```diff
@@ -0,0 +1,280 @@
+"""
+Prove an Ethereum block using Keth given a block number.
+Fetches zkpi data, converts it to EELS/Keth format, and generates a proof.
+"""
+
+import argparse
+import json
+import logging
+import sys
+from pathlib import Path
+from typing import Any, Dict, Optional
+
+from ethereum.cancun.blocks import Block, Withdrawal
+from ethereum.cancun.fork import BlockChain, apply_body, get_last_256_block_hashes
+from ethereum.cancun.fork_types import Address
+from ethereum.cancun.transactions import LegacyTransaction
+from ethereum.cancun.vm.gas import calculate_excess_blob_gas
+from ethereum.utils.hexadecimal import hex_to_bytes, hex_to_u256, hex_to_uint
+from ethereum_spec_tools.evm_tools.loaders.fixture_loader import Load
+from ethereum_types.bytes import Bytes, Bytes0, Bytes32
+from ethereum_types.numeric import U64, U256
+
+from cairo_addons.vm import run_proof_mode
+from tests.ef_tests.helpers.load_state_tests import convert_defaultdict
+
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+logger = logging.getLogger(__name__)
+
+CANCUN_FORK_BLOCK = 19426587  # First Cancun block
+
+
+def parse_args() -> argparse.Namespace:
+    """Parse command-line arguments."""
+    parser = argparse.ArgumentParser(description="Prove an Ethereum block using Keth")
+    parser.add_argument(
+        "block_number",
+        type=int,
+        help="Ethereum block number to prove (Cancun fork or later)",
+    )
+    parser.add_argument(
+        "--output-dir",
+        type=Path,
+        default=Path("output"),
+        help="Directory to save proof artifacts (default: ./output)",
+    )
+    parser.add_argument(
+        "--data-dir",
+        type=Path,
+        default=Path("data/1/eels"),
+        help="Directory containing ZKPI JSON files (default: ./data/1/eels)",
+    )
+    parser.add_argument(
+        "--compiled-program",
+        type=Path,
+        default=Path("build/main.json"),
+        help="Path to compiled Cairo program (default: ./build/main.json)",
+    )
+    parser.add_argument(
+        "--stwo-proof",
+        action="store_true",
+        help="Generate Stwo proof instead of traditional proof artifacts",
```

Comment:
this means: "generate the proof and not only the prover inputs" right?

Reply:
yes, generate the proof instead of storing prover inputs

Reply:
just reword as it can be understood as "stwo-kind artifacts instead of usual ones"